shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -315,9 +315,36 @@ public boolean onTouchEvent(MotionEvent event) {
       return false;
     }
 
+    // TODO(abarth): This version check might not be effective in some
+    // versions of Android that statically compile code and will be upset
+    // at the lack of |requestUnbufferedDispatch|. Instead, we should factor
+    // version-dependent code into separate classes for each supported
+    // version and dispatch dynamically.
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      requestUnbufferedDispatch(event);
+    }
+
     return androidTouchProcessor.onTouchEvent(event);
   }
 
+  /**
+   * Invoked by Android when a generic motion event occurs, e.g., joystick movement, mouse hover,
+   * track pad touches, scroll wheel movements, etc.
+   *
+   * Flutter handles all of its own gesture detection and processing, therefore this
+   * method forwards all {@link MotionEvent} data from Android to Flutter.
+   */
+  @Override
+  public boolean onGenericMotionEvent(MotionEvent event) {
+    boolean handled = isAttachedToFlutterEngine() && androidTouchProcessor.onGenericMotionEvent(event);
+    if (handled) {
```

Comment:
Couldn't you just do this?:
```suggestion
    return handled ? true : super.onGenericMotionEvent(event);
```
or get rid if the local variable and move the boolean into the if:
```suggestion
    if (isAttached() && androidTouchProcessor.onGenericMotionEvent(event)) {
```

Reply:
Went with your first option.